src/Dobble/Deck.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Marmelab\Dobble;
+
+/**
+ * @todo validate & generate functions
+ */
+class Deck implements \Countable
+{
+    private $cards = [];
+
+    public function __construct($cards = [])
+    {
+        if (!is_array($cards) && !$cards instanceof \Traversable) {
+            throw new \InvalidArgumentException('Not a list of cards');
+        }
+
+        foreach ($cards as $card) {
+            $this->append($card);
```

Comment:
You currently have no specific behavior in your `append` method. Keep it simple by just setting `$this->cards = $cards`.